### PR TITLE
rdiff-backup: 1.3.3 -> 2.0.5

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

Upstream update + move from py2 -> py3.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/backup/rdiff-backup/default.nix [hunk 1]
```diff
@@ -1,23 +1,36 @@
...
   doCheck = false;
 
-  meta = {
+  meta = with lib; {
     description = "Backup system trying to combine best a mirror and an incremental backup system";
-    homepage = "http://rdiff-backup.nongnu.org/";
-    license = lib.licenses.gpl2;
-    platforms = lib.platforms.all;
+    homepage = "https://rdiff-backup.net";
+    license = licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/tools/backup/rdiff-backup/default.nix:32]: Do you know if the license is gpl2Only or gpl2Plus?

File: pkgs/tools/backup/rdiff-backup/default.nix [hunk 2]
```diff
@@ -1,23 +1,36 @@
...
+  # pkg_resources fails to find the version and then falls back to "DEV"
+  postPatch = ''
+    substituteInPlace src/rdiff_backup/Globals.py \
+      --replace 'version = "DEV"' 'version = "${version}"'
+  '';
+
+  buildInputs = [ librsync ];
 
-  buildInputs = [ librsync gnused ];
+  nativeBuildInputs = with pypkgs; [ setuptools_scm ];
```
Comment (SuperSandro2000) [inline pkgs/tools/backup/rdiff-backup/default.nix:24]:
```suggestion
  nativeBuildInputs = with pypkgs; [ setuptools-scm ];
```
deprecated alias

File: pkgs/tools/backup/rdiff-backup/default.nix [hunk 3]
```diff
@@ -1,23 +1,36 @@
-{lib, fetchurl, python2Packages, librsync, gnused }:
+{ lib, python3Packages, librsync }:
 
-python2Packages.buildPythonApplication {
-  name = "rdiff-backup-1.3.3";
+let
+  pypkgs = python3Packages;
```
Comment (SuperSandro2000) [inline pkgs/tools/backup/rdiff-backup/default.nix:4]: If you add ``with python3.pkgs; `` to the callPackages function in all-packages.nix we can just use buildPythonApplication, fetchPypi, etc. as inputs.

**Comment by peterhoeg** [comment]
> Do you know if the license is gpl2Only or gpl2Plus?

Fixed, thanks.

> deprecated alias

Fixed, thanks.

> If you add with python3.pkgs; to the callPackages function in all-packages.nix we can just use buildPythonApplication, fetchPypi, etc. as inputs.

Pulling all of python3.pkgs into scope is an anti-pattern in my opinion (just like pulling in all of lib is). At the cost of having to be a bit more specific, it becomes clear where things are coming from.

**Pull request merged**